temporalio/client.py

```diff
@@ -1165,29 +1165,68 @@ async def report_cancellation(self, *details: Any) -> None:
         )
 
 
-class WorkflowDescription:
-    """Description for a workflow."""
-
-    def __init__(
-        self,
-        raw_message: temporalio.api.workflowservice.v1.DescribeWorkflowExecutionResponse,
-    ):
-        """Create a workflow description from a describe response."""
-        self._raw_message = raw_message
-        status = raw_message.workflow_execution_info.status
-        self._status = WorkflowExecutionStatus(status) if status else None
+@dataclass
+class WorkflowExecutionDescription:
+    """Description for a workflow execution."""
 
-    @property
-    def raw_message(
-        self,
-    ) -> temporalio.api.workflowservice.v1.DescribeWorkflowExecutionResponse:
-        """Underlying workflow description response."""
-        return self._raw_message
+    close_time: Optional[datetime]
+    execution_time: Optional[datetime]
+    history_length: int
+    id: str
```

Comment:
This should probably be `workflow_id`. It's that way in TS and you have to infer what it is by looking at run id and knowing there are two types of IDs, which seems less than ideal.

Reply:
In Python we have used `id` where it's obviously a workflow ID to avoid stuttering and unnecessary typing. So we have `WorkflowHandle.id`, `start_workflow(id="myid")` etc. This was how it was originally proposed/designed, but if we want to change all of these we can. I will bring it up.